udev/udev-configure-printer.c

```diff
@@ -1791,6 +1791,39 @@ disable_queue (const char *printer_uri, void *context)
   ippDelete (answer);
   httpClose (cups);
 }
+static int  
+compare_usb_uri(char *uri_store, char *uri_uevent)
+{
+  int index = 0;
+  int subset = 1;
+  int exist = -1; 
+  char *full_path = NULL;
+  struct stat buffer;
+
+  if (strlen(uri_uevent) < strlen(uri_store))
+    return -1; 
+
+  // check whether it's a substring
+  full_path = (char *) malloc(strlen(uri_store) + 4); 
+  memset(full_path, 0, sizeof(full_path));
+  strcpy(full_path, "/sys");
+  strcat(full_path, uri_store);
+  while (uri_store[index] != '\0'){
+    if (uri_store[index] != uri_uevent[index]){
+      subset = -1; 
+      break;
+    }   
+    index++;
+  }
+
+  // check the exist of this device
+  if (subset == 1 && stat(full_path, &buffer) == 0){ 
+   exist = 1;  
```

Comment:
I would propose to move it to a separate function - it is against the name of function. The name indicates it will compare the uris, not checking its existence.

```device_exists()``` can be an additional argument in ```if (compare_usb_uri())``` condition. IMO it causes no harm if it is run even on devices which aren't substrings, but they match will usb uri from udev event. What do you think?